gunicorn/reloader.py

```diff
@@ -26,7 +28,7 @@ def add_extra_file(self, filename):
 
     def get_files(self):
         fnames = [
-            re.sub('py[co]$', 'py', module.__file__)
+            COMPILED_EXT_RE.sub('py', module.__file__)
```

Comment:
The re module has internal caching, so I'm surprised moving the pattern to `re.compile()` had any effect. Out of curiosity, how did you measure the performance gain?

Reply:
It's true. I looked at cpython's `re` source to be sure. Maybe it's just the extra Python call stack clutter leading up to cache hit. I first noticed the issue after instrumenting workers with [profiling](https://github.com/what-studio/profiling) ([like this](https://github.com/what-studio/profiling/issues/31#issuecomment-397774043)). Isolated results with `timeit` and IPython's `%timeit`.

```python
import re
def old_way():
    return re.sub('py[co]$', 'py', 'stuff_and_things.pyc')

precompiled_pattern = re.compile(r'py[co]$')
def new_way():
    return precompiled_pattern.sub('py', 'stuff_and_things.pyc')
```
```
%timeit a=new_way()
The slowest run took 8.21 times longer than the fastest. This could mean that an intermediate result is being cached.
1000000 loops, best of 3: 584 ns per loop

%timeit a=old_way()
The slowest run took 6.37 times longer than the fastest. This could mean that an intermediate result is being cached.
1000000 loops, best of 3: 1.17 µs per loop
```